packages/mdc-slider/foundation.js

```diff
@@ -155,6 +212,9 @@ class MDCSliderFoundation extends MDCFoundation {
     this.setActive_(true);
     this.adapter_.activateRipple();
 
+    this.setInTransit_(!this.handlingThumbTargetEvt_);
```

Comment:
It seems like there are 2 cases when we want to add the `--in-transit` class to the slider:
1. When clicking on a part of the slider that's *NOT* the thumb or thumb sub-elements
2. When using the keydown arrow motions

If that's true (pls correct me if it's not!), then we can simplify the code a bit:
1. In `handleInteractionStart`, we can determine whether to add the `--in-transit` class based on what the click event target is.
```
handleInteractionStart(evt) {
  ...
  const shouldTransition = 
      this.adapter_.eventTargetHasClass(evt.target, 'mdc-slider') ||
      this.adapter_.eventTargetHasClass(evt.target, 'mdc-slider__track') ||
      this.adapter_.eventTargetHasClass(evt.target, 'mdc-slider__track-fill');
  this.setInTransit_(shouldTransition);
  ...
}
```
2. In `handleKeyDown`, always add the `--in-transit` class.
```
handleKeydown(evt) {
  ...
  this.setInTransit_(true);
  ...
}
```
3. Instead of [de]registering a `transitionend` handler in `updateUIForCurrentValue_()`, we can [de]register it along with all the other handlers in `init()` and `destroy()`. This means we'd check inside the transition end handler if `inTransit_` is true.
```
handleTransitionEnd(evt) {
  if (this.inTransit_ &&
      this.adapter_.eventTargetHasClass(evt.target, 'mdc-slider__track-fill')) {
    this.setInTransit_(false);
  }
}
```
In general it's a little easier to reason about when all handlers are registered in the same place. This also replaces the need for the `handlingThumbTargetEvt_` flag, the `handleThumbPointer` method, and the `[de]registerThumbEventHandler` adapter methods.

Reply:
Got it! Added and updated tests as well